internal/migrate/staging_validator.go

```diff
@@ -187,10 +190,12 @@ func (v *stagingValidatorImpl) Validate(stagedContracts []stagedContractUpdate)
 
 	v.stagedContracts = make(map[common.AddressLocation]stagedContractUpdate)
 	for _, stagedContract := range stagedContracts {
-		v.stagedContracts[stagedContract.DeployLocation] = stagedContract
+		stagedContractLocation := stagedContract.DeployLocation
```

Comment:
contract update validator relies on typeIDs. So had to use the `DeployLocation` in all places, instead of `SourceLocation`

Reply:
This is also needed to fix https://discord.com/channels/613813861610684416/1277718471618330707/1277750460723363911

Reply:
Added a dedicated test for ^above: https://github.com/onflow/flow-cli/pull/1705/commits/bd87b9fa478ee61027ea811f87bc8300d0136c1b